vue/src/modules/Workspace/article/attachment/Attachments.vue

```diff
@@ -0,0 +1,289 @@
+    <AttachmentLabel/>
+  </div>
+</template>
+
+<script>
+import {createNamespacedHelpers} from "vuex";
+import AttachmentLabel from "./AttachmentLabel";
+import AttachmentDeleteButton from "./AttachmentDeleteButton";
+import AttachmentDownloadButton from "./AttachmentDownloadButton";
+import AttachmentPreview from "./AttachmentPreview";
+
+const {mapState, mapActions} = createNamespacedHelpers('article');
+export default {
+  name: "Attachments",
+  components: {AttachmentPreview, AttachmentDownloadButton, AttachmentDeleteButton, AttachmentLabel},
+  data() {
+    return {
+      articleFileIds: [],
+      uploadedAttachmentIds: [],
+      progress: 0,
+      sortBy: null,
+      sortDesc: null
+    }
+  },
+  computed: {
+    ...mapState(['visibleColumns', 'articleFiles', 'attachConfig', 'currentArticle']),
+    fields() {
+      return [
+        {key: 'checkbox', label: ''},
+        {key: 'name', class: 'name', sortable: true},
+        ...this.visibleColumns.filter(c => c.visible),
+        {key: 'actions', class: 'text-center'}
+      ]
+    },
+    selected() {
+      return this.articleFiles.filter(a => a.selected === '1')
+    },
+    selectedIds() {
+      return this.selected.map(a => a.id)
+    },
+  },
+  watch: {
+    progress() {
+      if (this.progress >= 100) {
+        setTimeout(() => {
+          this.progress = 0;
+        }, 2000);
+      }
+    }
+  },
+  methods: {
+    ...mapActions(['getAttachConfig', 'attachFiles', 'showEditLabelDialog', 'showPreviewModal', 'sortAttachment']),
+    async onFileChoose(ev) {
+      const filesArray = [];
+      let filesName = [];
+
+      let maxFileSize = this.attachConfig.upload_max_filesize.size;
+      const maxFileUploads = parseInt(this.attachConfig.max_file_uploads.size);
+
+      let maxSizeInMb;
+      if (maxFileSize.includes('M')) {
+        maxSizeInMb = parseInt(maxFileSize.split('M')[0]);
+      } else if (maxFileSize.includes('G')) {
+        maxSizeInMb = parseInt(maxFileSize.split('G')[0]) * 1024;
+      }
+
+      let isFileTooBig = false;
+      let bigFile = '';
+      for (let i = 0; i < ev.target.files.length; i++) {
+        let fileSizeInMb = ev.target.files[i].size / 1000000;
+        filesArray.push(ev.target.files[i]);
+        filesName.push(ev.target.files[i].name);
+
+        if (fileSizeInMb > maxSizeInMb) {
+          isFileTooBig = true;
+          bigFile = ev.target.files[i].name
+        }
+      }
+      if (filesArray.length > maxFileUploads) {
+        this.$toast(this.$t(`Too much files for upload at same time`), 'danger');
+      } else if (isFileTooBig) {
+        this.$toast(this.$t(`File '{name}' is too big for upload`, {name: bigFile}), 'danger');
+      } else {
+        let checkFiles = this.checkFileNames(filesName);
+
+        if (checkFiles[0]) {
+          const result = await this.$confirm(this.$t(`${checkFiles.join(',')} file already exist. Are you sure that you want to overwrite them?`),
```

Comment:
@skavt  `${checkFiles.join(',')}`  this part should be dynamic for translation because it will be probably unique for all cases therefore writing them all in translations is not possible. Make something like this:
```
this.$confirm(`{file_names} file already exist. Are you sure that you want to overwrite them?`, {file_names: this.$t(`${checkFiles.join(',')}})
```